modules/utils/createUseScroll.js

```diff
@@ -27,13 +27,18 @@ export default function createUseScroll(updateScroll, start, stop) {
         }
       }
 
+      const { shouldUpdateScroll } = options
+      let oldLocation
       let listeners = [], currentLocation, unlisten
 
       function onChange(location) {
+        oldLocation = currentLocation
         currentLocation = location
 
         listeners.forEach(listener => listener(location))
-        updateScroll(location)
+        if (!shouldUpdateScroll || shouldUpdateScroll(oldLocation, currentLocation)) {
```

Comment:
Can you default this to a no-op function that just returns `true`? Makes the code read a bit better.